catalog/src/test/java/org/killbill/billing/catalog/TestDefaultPriceOverride.java

```diff
@@ -180,4 +184,192 @@ private void assertInternationalPrice(final InternationalPrice newInternationalP
+
+        final Matcher m = DefaultPriceOverride.CUSTOM_PLAN_NAME_PATTERN.matcher(overriddenPlan.getName());
+
+        assertTrue(m.matches());
+        assertEquals(m.group(1), plan.getName());
+        assertEquals(overriddenPlan.getProduct().getName(), plan.getProduct().getName());
+        assertEquals(overriddenPlan.getRecurringBillingPeriod(), plan.getRecurringBillingPeriod());
+        if (plan.getEffectiveDateForExistingSubscriptions() != null) {
+            assertEquals(overriddenPlan.getEffectiveDateForExistingSubscriptions().compareTo(plan.getEffectiveDateForExistingSubscriptions()), 0);
+        }
+        assertNotEquals(overriddenPlan.getFinalPhase().getName(), plan.getFinalPhase().getName());
+
+        for(int i = 0 ; i < overriddenPlan.getFinalPhase().getUsages().length; i++) {
+            final DefaultUsage initialUsage = (DefaultUsage) plan.getFinalPhase().getUsages()[i];
+            final DefaultUsage newUsage = (DefaultUsage) overriddenPlan.getFinalPhase().getUsages()[i];
+
+            assertEquals(newUsage.getName(), initialUsage.getName());
+            assertEquals(newUsage.getUsageType(), initialUsage.getUsageType());
+            assertEquals(newUsage.getBillingPeriod(), initialUsage.getBillingPeriod());
+            assertEquals(newUsage.getBillingPeriod(), initialUsage.getBillingPeriod());
+            assertEquals(newUsage.getTiers().length, initialUsage.getTiers().length);
+
+            for(int j = 0 ; j < newUsage.getTiers().length; j++){
+                final DefaultTier initialTier = (DefaultTier) initialUsage.getTiers()[j];
+                final DefaultTier newTier = (DefaultTier) newUsage.getTiers()[j];
+                assertEquals(newTier.getTieredBlocks().length, initialTier.getTieredBlocks().length);
+
+                for(int k = 0; k < newTier.getTieredBlocks().length; k++){
+                    final DefaultTieredBlock initialTieredBlock = (DefaultTieredBlock) initialTier.getTieredBlocks()[k];
+                    final DefaultTieredBlock newTieredBlock = (DefaultTieredBlock) newTier.getTieredBlocks()[k];
+                    final TieredBlockPriceOverride override = Iterables.tryFind(tieredBlockPriceOverrides, new Predicate<TieredBlockPriceOverride>() {
+                        @Override
+                        public boolean apply(final TieredBlockPriceOverride input) {
+
+                            return input.getUnitName().equals(initialTieredBlock.getUnit().getName()) &&
+                                    Double.compare(input.getSize(), initialTieredBlock.getSize()) == 0 &&
+                                    Double.compare(input.getMax(), initialTieredBlock.getMax()) == 0;
+                        }
+                    }).orNull();
+
+                    assertEquals(newTieredBlock.getUnit().getName(), initialTieredBlock.getUnit().getName());
+                    assertEquals(newTieredBlock.getMax(),newTieredBlock.getMax());
```

Comment:
Seems like there are some typos here. Shouldn't we check `newTieredBlock` against `initialTieredBlock` in all assertions?